sotodlib/hwp/g3thwp.py

```diff
@@ -1059,12 +1062,21 @@ def _irig_quality_check(self):
             self._irig_time = np.array([])
             self._rising_edge = np.array([])
 
+    def _process_counter_overflow_glitch(self):
+        """ Treat glitches due to 32 bit internal counter overflow """
+        idx = np.where((np.diff(self._encd_clk)>2**32) & (np.diff(self._encd_clk)<2**32+1e6))[0] + 1
```

Comment:
This is surprising to me ... doesn't `_encd_clk` increase at each step?  So the overflow should be when diff is a large *negative* number?

Also, a typical overflow in a 32-bit truncated word would not have a diff as big as 2**32 ... it would only be 2**32 - 1, and it's often simply detected by comparing to 2**31.

So if what you put here is correct, maybe it needs more comments describing the issue that you are solving with this...

Reply:
I was surprized to find this glitch too. 
I modified the detection criteria.
To be honest, I'm not exactly sure why this happens, but I found that the 32-bit jump can explain the glitch.

The bbb internal counter overflows roughly once in 35 mins (200MHz, 32bit). 
This jump doesn't happen most of the time.